sdk/core/azure-core/CHANGELOG.md

```diff
@@ -9,6 +9,11 @@
   - Added `HttpRequest.getBodyAsBinaryData()`.
   - Added `HttpRequest.setBody(BinaryData)`.
   - Added `BinaryData.fromFlux(Flux<ByteBuffer>, Long, boolean)` that allows both buffered and non-buffered handling of `Flux<ByteBuffer>`.
+- Added support for sending synchronous requests using `sendSync` in `HttpPipeline`:
+  - Added `HttpPipelinePolicy.processSync(HttpPipelineCallContext context, HttpPipelineNextSyncPolicy next)` to allow processing policies synchronously.
+  - Added `HttpPipelineSynchronousPolicy` to represent synchronous `HttpPipelinePoicy`.
```

Comment:
Initial gut reaction is that this policy should use the `Sync` naming, to be consistent with other `Sync` classes and API. I'm also curious if the more appropriate name is to move the `Sync` to be a prefix, so we have `SyncHttpPipelinePolicy` (and, for the other type referenced below, `SyncHttpPipelineNextPolicy`).

Reply:
I think that there's more value in suffix than prefix. I.e. `HttpPipelineNextPolicy` and `HttpPipelineNextPolicySync` should show up next to each other in IDE hints or javadoc listings. `HttpPipelineNextPolicy` and `HttpPipelineNextSyncPolicy` also land closely in places that list types. 
FWIW. when it comes to `Async` we have `DigitalTwinsClient` and `DigitalTwinsAsyncClient` - i.e. the `Async` goes before last noun. If we were to reinvent it would we go with `DigitalTwinsSyncClient`/`DigitalTwinsClient` or  `SyncDigitalTwinsClient`/`DigitalTwinsClient`, `DigitalTwinsClientSync`/`DigitalTwinsClient` ?
Btw. if we were starting from scratch we'd probably have `HttpPipelineNextPolicy` and `HttpPipelineNextAsyncPolicy` to match what we do with clients today.

As for the `HttpPipelineSynchronousPolicy`/`SyncHttpPipelinePolicy` - I thin we need to come up with better name. Maybe `HttpPipelineNonBlockingPolicy` or `HeadersOnlyHttpPipelinePolicy` (and only pass headers to before/after extension points instead of full req/res to fortify this further).

Reply:
`HttpPipelineSyncPolicy ` is the pattern we follow for naming types as @kasobol-msft  suggested above and it's also good to have IntelliSense show these two types together.